thegame/engine/game_objects/game_object.py

```diff
@@ -1,7 +1,46 @@
 class GameObject:
     """ An object that will be used to fill the objects sheet of a game."""
 
-    def __init__(self, sprite, animation=None):
+    def __init__(self, sprite_location: str, animation: str = None, name: str = None):
 
-        self.sprite = sprite
+        self.sprite_location = sprite_location
         self.animation = animation
+        self._loaded_sprite = None
+        self.name = name
+
+    def clone(self):
+        clone = GameObject(
+            sprite_location=self.sprite_location, animation=self.animation
+        )
+
+        return clone
+
+    def register_loaded_sprite(self, loaded_sprite):
+        self._loaded_sprite = loaded_sprite
+
+    def set_sprite_position(self, pos_x, pos_y):
+        if self._loaded_sprite is None:
+            raise ValueError(
```

Comment:
Should this be a ValueError?

Reply:
No, this is an issue with an attribute being unset, and therefore should be an AttributeError

Reply:
Addressed with https://github.com/Ross-Alexandra/TheGame/pull/16/commits/1711bd575b771a88304b64123375c7bdb693ac66